app/Livewire/Profile.php

```diff
@@ -41,8 +64,58 @@ public function save()
         }
         $user = User::findOrFailByUsername($this->uid);
         $user->setAttribute('mail', $this->email);
-        $user->setAttribute('cn', $this->fullName);
+        $user->setAttribute('givenName', $this->givenName);
+        $user->setAttribute('sn', $this->sn);
+        $user->setAttribute('cn', $this->givenName . ' ' . $this->sn);
+        $user->setAttribute('description', $this->course);
+        $user->setAttribute('street', $this->street);
+        $user->setAttribute('postalCode', $this->postalCode);
+        $user->setAttribute('l', $this->city);
+        $user->setAttribute('telephoneNumber', $this->phone);
+        $user->setAttribute('mobile', $this->mobile);
         $user->save();
         return redirect()->route('profile')->with('message', __('Saved'));
     }
+
+    public function savePicture()
+    {
+        $img = imagecreatefromstring(base64_decode(str_replace('data:image/jpeg;base64,', '', $this->picture))); // convert base64 string to image object
+        $imgSize = 400; // size the image should be resized to
+        $imgFileName = Str::uuid() . '.jpg'; // generate a file name
+        $width = imagesx($img); // initial width of the image
+        $height = imagesy($img); // initial height of the image
+
+        // Resize the image
+        $thumb = imagecreatetruecolor($imgSize, $imgSize);
+        imagecopyresized($thumb, $img, 0, 0, 0, 0, $imgSize, $imgSize, $width, $height);
+        ob_start();
+        imagejpeg($thumb, NULL);
+        $imgResized = ob_get_clean();
+
+        // Put resized image into Laravel Storage
+        Storage::disk('public')->put('pictures/' . $imgFileName, $imgResized, 'public');
+
+        // Set picture URL
+        $this->pictureUrl = Storage::disk('public')->url('pictures/' . $imgFileName);
+
+        // Write image URL to LDAP
+        $user = User::findOrFailByUsername($this->uid);
+        $user->setAttribute('jpegPhoto', $this->pictureUrl);
+        $user->save();
+
+        return redirect()->route('profile')->with('message', __('Saved'));
+    }
+
+    public function deletePicture()
+    {
+        // Remove image from Laravel Storage
+        Storage::disk('public')->delete(str_replace(env('APP_URL') . '/storage/', '', $this->pictureUrl));
```

Comment:
`env(...)` sollte nur in config files verwendet werden. Außerhalb sollte `config(...)`  verwendet werden, sonst kann man den config cache nicht vewenden. See https://laravel.com/docs/12.x/configuration#accessing-configuration-values